src/lib/desmoHub-module.ts

```diff
@@ -1,325 +1,374 @@
+ */
+  public async getNewRequestID(): Promise<void> {
+    if (!this.isConnected) {
+      throw new Error(
+        'This method requires the wallet signer to be already signed-in!',
+      );
+    }
+
+    const tx = await this.contract.getNewRequestID();
+    this.TRANSACTION_SENT.next({
+      invokedOperation: OperationType.getNewRequestID,
+      hash: tx.hash,
+      sent: new Date(Date.now()),
+    });
+  }
+/**
+ * 
+ * @returns the length of the TDD storage.
+ */
+  public async getTDDStorageLength(): Promise<ethers.BigNumber> {
+    return await this.contract.getTDDStorageLength();
+  }
+/**
+ * Get the scores relative to a request id.
+ * 
+ * @param requestID 
+ * @returns 
+ */
+  public async getScoresByRequestID(
+    requestID: ethers.Bytes,
+  ): Promise<ethers.Bytes> {
+    return await this.contract.getScoresByRequestID(requestID);
+  }
+/**
+ * Get TDD by request id.
+ * 
+ * @param requestID 
+ * @returns 
+ */
+  public async getTDDByRequestID(requestID: ethers.Bytes): Promise<string[]> {
+    return await this.contract.getTDDByRequestID(requestID);
+  }
+
+  /**
+   * 
+   * @returns the TDD of the current user.
+   */
+  public async getTDD(): Promise<ITDD> {
+    const { url, owner, disabled, score }: ITDD = await this.contract.getTDD();
+    return { url, owner, disabled, score } as ITDD;
+  }
+
+  /**
+   * @param start the initial TDD index to retrieve
+   * @param stop the index at which to stop retrieving TDDs (if not specified, the full length of the TDD storage inside the contract)
+   * @returns an array containing the requested TDD descriptions.
+   */
+  public async getTDDList(start = 0, stop?: number): Promise<ITDD[]> {
+    const tddStorageLength = (await this.getTDDStorageLength()).toNumber();
+
+    if (tddStorageLength < 1) {
+      throw new Error('No TDDs are currently registered in the contract.');
+    }
+
+    // Make sure that `start` and `stop` are integer numbers:
+    start = Math.floor(start ?? 0);
+    stop = Math.floor(stop ?? tddStorageLength);
+
+    // Apply limits to `start` and `stop` indexes:
+    start = Math.max(start, 0);
+    stop = Math.min(stop, tddStorageLength);
+
+    if (start >= tddStorageLength) {
+      throw new Error(`Start index must be lower than the TDD storage length (${tddStorageLength}).`);
+    } else if (start > stop) {
+      throw new Error(
+        `Start index (${start}) is greater than stop index (${stop}).`,
+      );
+    }
+
+    /**
+     * Summarizing, here we should have:
+     * 0 <= start < tddStorageLength
+     * 0 <= stop <= tddStorageLength
+     */
+
+    const promises: Promise<ITDD>[] = [];
+    for (let i = start; i < stop; ++i) {
+      promises.push(this.contract.getTDDByIndex(i));
+    }
+    const tddList = await Promise.all(promises);
+
+    // Clean up the TDD objects from their numeric properties:
+    for (let i = 0; i < tddList.length; ++i) {
```

Comment:
```suggestion
    for (let i = 0; i < tddList.length; i++) {
```
a more mainstream for-loop.

Reply:
Not needed because we chose to replace the for loop with a call to `map()`.